python/mxnet/numpy/multiarray.py

```diff
@@ -148,10 +148,10 @@ def _reshape_view(a, *shape):  # pylint: disable=redefined-outer-name
 
 def _as_mx_np_array(object, ctx=None):
     """Convert object to mxnet.numpy.ndarray."""
-    if isinstance(object, _np.ndarray):
-        if not object.flags['C_CONTIGUOUS']:
-            object = _np.ascontiguousarray(object, dtype=object.dtype)
```

Comment:
Should we make sure it's c_contiguous?

Reply:
I'm changing it to make a copy of the numpy array so it's already handled by the usage of `array()`. The reason for not using zero-copy `from_numpy` is that `from_numpy` changes the original array to be read-only, because mxnet execution is asynchronous and thus mutation from numpy side on the same buffer will result in undefined behavior.